pilot/pkg/model/push_context.go

```diff
@@ -2163,10 +2163,15 @@ func (ps *PushContext) ServiceAccounts(hostname host.Name, namespace string, por
 }
 
 func (ps *PushContext) SupportsTunnel(ip string) bool {
```

Comment:
nit; since tunnel is a property of a workload instead of a service, I think this function should just take a workload instead of an address. then we can remove the TODO and service related logic and switch as well

Reply:
this doesn't actually work, the caller is a string type not an `Address` proto.